portfolio/src/main/webapp/comment.js

```diff
@@ -1,28 +1,113 @@
 
-/** Creates an <li> element containing text. Use this to show a list of comments for now. */
+/** Creates an <p> element containing text. Use this to show a list of comments for now. */
 function createListElement(text) {
-  const liElement = document.createElement('li');
-  liElement.innerText = text;
+  const liElement = document.createElement('p');
+  console.log('comment text ' + text);
+  var nameComment = makeNameCommentArray(text);
+  liElement.innerText = nameComment[1];
+  liElement.style.paddingLeft = "15px";
+  liElement.style.fontSize = "15px";
   return liElement;
 }
 
+function makeNameCommentArray(text) {
+  var nameComment = text.split(": ");
+  return nameComment;
+}
+
+function createNameElement(text) {
+  var nameComment = makeNameCommentArray(text);
+  var name = nameComment[0];
+  console.log('NAME ' + name);
+  const nameElement = document.createElement('p');
+  nameElement.innerText = name;
+  nameElement.style.paddingLeft = "15px";
+  nameElement.style.fontSize = "18px";
+  nameElement.style.fontWeight = "bold";
+  return nameElement;
+}
+
 /** Make an unordered list of all comments in the server, print list onto page. */
 function makeCommentList() {
   console.log('Get comment function.');
 
+  // Make string representing the URL with the number of comments selected by user.
   const str = '/data?number-comments=' + document.getElementById('number-comments').value;
   console.log(str);
 
-  //todo replace the 6 with the value of the html element
-  fetch(str).then(response => response.json()).then((commentObj) => {
+  listComments(str);
+}
+
+function listComments(address) {
+  fetch(address)
+  .then(response => response.json()) // Convert to json
+  .then((commentObj) => {
     const commentSpace = document.getElementById('comment-space');
     
+    // Clear out comments from element.
     while (commentSpace.firstChild) {
       commentSpace.removeChild(commentSpace.firstChild);
     }
 
     for (i in commentObj) {
```

Comment:
I haven't used javascript in a while but this syntax looks odd. Is `commentObj` an array or an object? If it is an array, is `i` the index or is it an entry in the array? If `commentObj` is an object like the name suggests, then `i` is a field in `commentObj`?

We can use naming to make this clearer. `i` doesn't give a hint as to what it is and `commentObj` makes it seem like it is an object but then the for concept is really weird.

Reply:
This was one of my confusions regarding objects in javascript. I now realize commentObj is an array of strings, so I'll rename it to commentArray or comments? Also, I believe i is the index. I will rename that to be called index.